src/components/OwnerMessage/OwnerMessage.jsx

```diff
@@ -3,7 +3,8 @@ import { useState, useEffect } from 'react';
 import { toast } from 'react-toastify';
 
 import { Button, Modal, ModalHeader, ModalBody, ModalFooter, Input } from 'reactstrap';
-import { connect } from 'react-redux';
+import { connect, useDispatch } from 'react-redux';
```

Comment:
```suggestion
import { connect } from 'react-redux';
```
Connect is already imported and used, so useDispatch is not needed. Can instead use `mapDispatchToProps` ( add `hasPermission: (permission) => dispatch(hasPermission(permission))` to `mapDispatchToProps`) and change `dispatch(hasPermission(...))` to `props.hasPermission(...)`.

I don't think it would have any functionality difference, but might be a bit cleaner/more constant in how this file is dispatching.